Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Network_Security_Group_Remediation.yml

```diff
@@ -0,0 +1,1382 @@
+    type: regular
+    task:
+      id: 110e9510-f313-4536-8113-59b63fad02fc
+      version: -1
+      name: Set variable for existing allow rule name
+      description: Set a variable for the existing allow rule.
+      scriptName: Set
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "32"
+    scriptarguments:
+      key:
+        simple: existingAllowRule
+      value:
+        complex:
+          root: AzureNSG.Rule
+          filters:
+          - - operator: isEqualString
+              left:
+                value:
+                  simple: AzureNSG.Rule.direction
+                iscontext: true
+              right:
+                value:
+                  simple: Inbound
+          - - operator: isEqualString
+              left:
+                value:
+                  simple: AzureNSG.Rule.access
+                iscontext: true
+              right:
+                value:
+                  simple: Allow
+          - - operator: containsGeneral
+              left:
+                value:
+                  simple: AzureNSG.Rule.name
+                iscontext: true
+              right:
+                value:
+                  simple: remediation-allow-port
+          - - operator: containsGeneral
+              left:
+                value:
+                  simple: AzureNSG.Rule.name
+                iscontext: true
+              right:
+                value:
+                  simple: ${inputs.RemotePort}
+                iscontext: true
+          - - operator: containsGeneral
+              left:
+                value:
+                  simple: AzureNSG.Rule.name
+                iscontext: true
+              right:
+                value:
+                  simple: ${inputs.RemoteProtocol}
+                iscontext: true
+          - - operator: notContainsGeneral
+              left:
+                value:
+                  simple: AzureNSG.Rule.destinationAddressPrefix
+                iscontext: true
+              right:
+                value:
+                  simple: ${inputs.AzureVMPrivateIP}
+                iscontext: true
+    separatecontext: false
+    continueonerrortype: ""
+    view: |-
+      {
+        "position": {
+          "x": 250,
+          "y": 930
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "32":
+    id: "32"
+    taskid: 353ca115-d23d-4e9c-893b-576dcf8e4a49
+    type: regular
+    task:
+      id: 353ca115-d23d-4e9c-893b-576dcf8e4a49
+      version: -1
+      name: Update existing remediation allow rule
+      description: |-
+        Update an existing security allow rule that denies all internet traffic to exposed public IPs (using the private IP) with an additional public IP (using the private IP). If one does not exist, it will be created.
```

Comment:
Should the definition for allow rule (as stated here) be the same as for deny rule (on line 732).